internals/cli/env.go

```diff
@@ -266,3 +353,80 @@ func splitVar(prefix, separator, envVar string) (string, string, bool) {
 	prefix = fmt.Sprintf("%s%s", strings.ToUpper(prefix), separator)
 	return split[0], split[1], strings.HasPrefix(strings.ToUpper(split[0]), prefix)
 }
+
+type ArgValue interface {
+	Set(string) error
+}
+
+func ArgumentRegister(params []ArgValue, args []string) error {
+	for i, arg := range args {
+		err := params[i].Set(arg)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type ArgArrValue interface {
+	Set([]string) error
+}
+
+type StringArgValue struct {
+	Param string
+}
+
+func (s *StringArgValue) Set(replacer string) error {
+	s.Param = replacer
+	return nil
+}
+
+type StringArrArgValue struct {
+	Param []string
+}
+
+func (s *StringArrArgValue) Set(replacer []string) error {
+	copy(s.Param, replacer)
+	return nil
+}
+
+type URLArgValue struct {
+	Param *url.URL
+}
+
+func (s *URLArgValue) Set(replacer string) error {
+	var err error
+	s.Param, err = url.Parse(replacer)
+	return err
+}
+
+// Registerer allows others to register commands on it.
+type Registerer interface {
+	Command(cmd string, help string) *CommandClause
+}
+
+// BindAction binds a function to a command clause, so that
+// it is executed when the command is parsed.
+func (c *CommandClause) BindArguments(params []ArgValue) {
```

Comment:
This might be a little bit trickier than we expected.
After the due research, I think that the best solution would be to include the arguments' descriptions in the help text of the command they belong to.
The problem that I'm referring is something similar to what's being talked about here: https://github.com/spf13/cobra/issues/571

However, I will continue to try a few other things, and will come back with updates as I go through the ideas I currently have.

Reply:
Placeholders and descriptions for arguments have been added now.